test/run_test.py

```diff
@@ -509,15 +509,15 @@ def run_test(
         return 0
 
     if is_cpp_test:
-        stepcurrent_key = test_file
+        stepcurrent_key = f"{test_file}_{os.urandom(8).hex()}"
     else:
         unittest_args.extend(
             [
                 f"--shard-id={test_module.shard - 1}",
                 f"--num-shards={test_module.num_shards}",
             ]
         )
-        stepcurrent_key = f"{test_file}_{test_module.shard - 1}"
+        stepcurrent_key = f"{test_file}_{test_module.shard - 1}_{os.urandom(8).hex()}"
```

Comment:
Does stepcurrent plugin still work after adding the randomized suffix?  The next run will need to be able to pickup the filename via this key and each run will have a different filename, so I'm thinking `conftest.py` might need to be updated too to handle the new logic?

Reply:
I want the same run_test call to use the same stepcurrent key, but different ones to use different stepcurrent keys.  File level retries happen within a run_test call, so they should all be using the same stepcurrent key, but if someone calls run_test again, it should be different, like the case with distributed.

Reply:
I see.  So file level retries are spawned within the same run_test process, it's good then